pkg/operator/staticpod/README.md

```diff
@@ -0,0 +1,161 @@
+# Static Pod Controllers
+
+OpenShift 4.x deploys the following foundational components as [static
+pods](https://kubernetes.io/docs/tasks/configure-pod-container/static-pod/):
+
+- etcd
+- kube-apiserver
+- kube-controller-manager
+- kube-scheduler
+
+These components comprise the minimum control plane on which all other
+OpenShift components are hosted.
+
+The operators that manage these foundational components rely on the
+set of controllers found in the `pkg/operator/staticpod` package and
+its children. These controllers are collectively responsible for
+generating static pod manifests and related configuration and writing
+them to the static pod manifests (`/etc/kubernetes/manifests`) and
+static pod resources (`/etcd/kubernetes/static-pod-resources`) paths,
+respectively, on each control plane node.
+
+The controllers take as input resources read from the OpenShift
+API. That means these controllers won't work without an
+apiserver. Installation of a new cluster involves creating bootstrap
+versions of the foundational components before these controllers can
+take over.
+
+## High-level flow
+
+A given operator (e.g. `cluster-kube-apiserver-operator`) configures a
+set of static pod controllers via the `staticpod` package's
+`NewBuilder` function and associated `With*` methods. Configuration includes:
+
+ - the set of configmaps and secrets containing configuration data for
+   the static pod
+   - the first configmap is expected to contain the static pod
+ - the set of configmaps and secrets containing ca bundles and tls certs
+   - resources containing tls data are supplied separately from other
+     resources to allow reload without restart/redeploy of the static pod
+ - the operator namespace
+
+A static pod rollout occurs as follows:
+
+ - the revision controller determines that a new revision is
```

Comment:
forgot to indicate that a revision is a copy of the configmaps and secrets specified in the configuration.